File: backend/app/models/vehicle_document.py
import uuid
from datetime import date

from sqlalchemy import Date, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base
from app.models.base import TimestampMixin, UUIDPKMixin


class VehicleDocument(Base, UUIDPKMixin, TimestampMixin):
    __tablename__ = "vehicle_documents"

    company_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("companies.id", ondelete="CASCADE"), nullable=False
    )
    vehicle_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("vehicles.id", ondelete="CASCADE"), nullable=False
    )
    document_type_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("vehicle_document_types.id"), nullable=False
    )
    number: Mapped[str | None] = mapped_column(String(100), nullable=True)
    expiry_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)

    document_type: Mapped["VehicleDocumentType"] = relationship()

File: frontend/src/types/vehicle_document.ts
export interface VehicleDocumentType {
  id: string
  company_id: string
  name: string
  alert_days_before: number
  created_at: string
  updated_at: string
}

export interface VehicleDocument {
  id: string
  company_id: string
  vehicle_id: string
  document_type_id: string
  number: string | null
  expiry_date: string | null
  notes: string | null
  created_at: string
  updated_at: string
  document_type: VehicleDocumentType
}
